### DISCUSSION TITLE
Why increasing the step/resolution produces holes in my graph?

### DISCUSSION THREAD

**Discussion by AntoineThebaud** [thanos-io/thanos#5418] [Questions & Answers]
Thanos version: `0.26.0`

Hello, there is a behavior I don't understand with Thanos: I'm trying to display a rate over a long period of time (30d), using only raw data.

With rate(5m) & a 5m resolution, all fine:
![image](https://user-images.githubusercontent.com/7058693/173385200-8f6fca9a-6808-471d-a26f-216c09da126d.png)

However with rate(5m) and 30m resolution:
![image](https://user-images.githubusercontent.com/7058693/173385607-11bccdba-eccc-4714-8078-e55e255a4fd9.png)
(It's working fine the 3 first days because prometheus is the master source of truth on this period)

Displaying a rate(5m) every 30m is not a good idea anyway, but my real query is actually using max_over_time(30m) with a rate subquery & I face the same problem:
![image](https://user-images.githubusercontent.com/7058693/173386073-9dd019d8-2507-44d3-8ac2-5f5920ea294f.png)

Is this behavior expected? I get the exact same result on Grafana UI btw. I thought the resolution param (step on Grafana) was a matter of FE display only, since I assumed "only raw data" was informing Thanos to use as much datapoints as it can for the computation (i.e every 1min here). But since the rate fails it seems increasing the resolution actually increases the gap between each point in the computation at some point, hence the fail?

More precisely, with this example, there is a clear edge at 25m resolution (it's working fine until 24m59s)
![image](https://user-images.githubusercontent.com/7058693/173389047-7ab93368-d9fc-45ff-b1f3-7541ceec46f7.png)
![image](https://user-images.githubusercontent.com/7058693/173389275-90589aed-3ea1-4cfc-9016-015a6f009fe8.png)

_Sorry in advance if this already answered/documented somewhere but in such case I didn't find the info._

**Comment by ahurtaud** [comment]
After more investigations, the issue happens only when we go through the query-frontend component, and also only data stored in object storage and provided by thanos store as sidecar data is working as you can see on previous screen captures...
When querying thanos query directly (no frontend), the exact same query with step=1500 works for the full range like 1499s ...
any idea ??
